lib/pbench/test/unit/server/test_upload.py

```diff
@@ -493,3 +512,68 @@ def setvalue(dataset: Dataset, key: str, value: Any):
         assert audit[1].user_name == "drb"
         assert audit[1].reason == AuditReason.INTERNAL
         assert audit[1].attributes == {"message": "INTERNAL ERROR"}
+
+    @pytest.mark.freeze_time("1970-01-01")
+    def test_upload_archive(
+        self, client, pbench_drb_token, server_config, setup_ctrl, tarball
+    ):
+        """Test a successful archiveonly dataset upload."""
+        datafile, _, md5 = tarball
+        with datafile.open("rb") as data_fp:
+            response = client.put(
+                self.gen_uri(server_config, datafile.name),
+                data=data_fp,
+                headers=self.gen_headers(pbench_drb_token, md5),
+                query_string={"metadata": "server.archiveonly:t,server.origin:test"},
+            )
+
+        assert response.status_code == HTTPStatus.CREATED, repr(response.data)
+        name = Dataset.stem(datafile)
+
+        dataset = Dataset.query(resource_id=md5)
+        assert dataset is not None
+        assert dataset.resource_id == md5
+        assert dataset.name == name
+        assert dataset.uploaded.isoformat() == "1970-01-01T00:00:00+00:00"
+        # assert Metadata.getvalue(dataset, Metadata.ARCHIVEONLY) is True
```

Comment:
Why is line 538 commented out?  (Should it be uncommented or removed?)

(And, FWIW, I believe that PEP recommends against explicit tests for boolean value, since the result already gets evaluated in a boolean context; similarly, at line 534, we can simply `assert dataset` to ensure that it is not `None`, and that also covers other, weird cases, as well.)

Reply:
I was debugging this after I had some problems. Now I have to wonder if I abandoned the debugging too quickly. Hmm...